Python/tier2_redundancy_eliminator_bytecodes.c

```diff
@@ -197,6 +197,17 @@ dummy_func(void) {
         }
     }
 
+    op(_BINARY_OP_ADD_UNICODE, (left, right -- res)) {
+        if (is_const(left) && is_const(right)) {
+            PyObject *temp = PyUnicode_Concat(get_const(left), get_const(right));
+            ERROR_IF(temp == NULL, error);
+            OUT_OF_SPACE_IF_NULL(res = sym_new_const(ctx, temp));
+        }
+        else {
+            OUT_OF_SPACE_IF_NULL(res = sym_new_known_type(ctx, &PyUnicode_Type));
```

Comment:
FWIW I am still in favor of splitting all calls to `OPT_OF_SPACE_IF_NULL()` into two, one with the side effect (`res = ...` etc.) and one with the error check. Slightly more verbose but doesn't bury the lede.